### PR TITLE
fix siglanCb logic

### PR DESCRIPTION
**Pull request by peeech**

### CODE REVIEW

**Review by zippy — APPROVED** [review]

**Review by timotree3 — APPROVED** [review]

**Comment by Connoropolous** [comment]
just for reference, this is the same as #42 so close that one

File: src/websocket/client.ts [hunk 1]
```diff
@@ -76,18 +76,22 @@ export class WsClient {
...
+              const decodedPayload = signalTransform(decodedMessage.App[1]);
 
-            // Return a uniform format to UI (ie: { type, data } - the same format as with callZome and appInfo...)
-            const signal: AppSignal = { type: msg.type , data: { cellId: decodedCellId, payload: decodedPayload }};
-            signalCb(signal);
+              // Return a uniform format to UI (ie: { type, data } - the same format as with callZome and appInfo...)
+              const signal: AppSignal = { type: msg.type , data: { cellId: decodedCellId, payload: decodedPayload }};
+              signalCb(signal);
+            } else {
+              console.error(`Received signal but no signal callback was set in constructor`);
```
Comment (Connoropolous) [inline src/websocket/client.ts]: should this be a console.error? not setting a signalCb seems like a common use possibility

Comment (peeech) [reply src/websocket/client.ts]: the way I see it is:
- if holochain is sending signals and there's no signalCB then it is an error
So not setting a signalCb in an application that does not use signals is fine and will never log an error

thoughts?

Comment (Connoropolous) [reply src/websocket/client.ts]: Mostly fair point. One more thing to consider is that for a given holochain conductor, you wish (and I have done this) to construct multiple clients at different points and places in the code. Some may attach the callback listener, and provide the one and only handler, and others may not. The one that doesn't will log this error despite you having another one which is handling them

Comment (peeech) [reply src/websocket/client.ts]: Yes, I see. So maybe a better solution would be to `console.log` it once during existence of this class instance? Easy to do and quite elegant I think.

Comment (timotree3) [reply src/websocket/client.ts]: That sounds great!

Comment (Connoropolous) [reply src/websocket/client.ts]: Yep!

Comment (peeech) [reply src/websocket/client.ts]: cf5cb68 should do the job

File: src/websocket/client.ts [hunk 2]
```diff
@@ -12,10 +12,12 @@ import { AppSignal, AppSignalCb, SignalResponseGeneric } from '../api/app'
 export class WsClient {
   socket: Websocket
   pendingRequests: Record<string, { fulfill: Function }>
+  noSignalCbFlag: Boolean
```
Comment (timotree3) [inline src/websocket/client.ts]: Suggesting a longer variable name and changing the type from `Boolean` (a class) to `boolean` (typescript primitive)
```suggestion
  alreadyWarnedNoSignalCb: boolean
```

**Review by timotree3 — APPROVED** [review]
Newest commit looks good and I have a couple of suggestions should you choose to accept them

**Comment by timotree3** [comment]
You're gonna need to update the usages of `noSignalCbFlag` to the new name as well

File: src/websocket/client.ts [hunk 3]
```diff
@@ -76,18 +78,24 @@ export class WsClient {
...
+              const decodedCellId = decodedMessage.App[0];
+              // Note:In order to return readible content to the UI, the signal payload must also be decoded.
+              const decodedPayload = signalTransform(decodedMessage.App[1]);
+
+              // Return a uniform format to UI (ie: { type, data } - the same format as with callZome and appInfo...)
+              const signal: AppSignal = { type: msg.type , data: { cellId: decodedCellId, payload: decodedPayload }};
+              signalCb(signal);
+            } else {
+              if (!hw.noSignalCbFlag) console.log(`Received signal but no signal callback was set in constructor`);
+              hw.noSignalCbFlag = true;
```
Comment (timotree3) [inline src/websocket/client.ts]: Two more variable usages to update
```suggestion
              if (!hw.alreadyWarnedNoSignalCb) console.log(`Received signal but no signal callback was set in constructor`);
              hw.alreadyWarnedNoSignalCb = true;
```

**Review by Connoropolous — APPROVED** [review]

**Pull request merged**